### PR TITLE
Begin split to sub-projects, move current project to a single sub-project

### PR DESCRIPTION
**Pull request by DanVanAtta**
Moves existing source into a submodule called "game-core"

- [x] TODO: Gradle checkstyle task does not working, looks to be checking nothing and just reports success
- [x] Fix integ test
- [x] See if we can keep checkstyle file where it is
- [x] Fix source code merging issues, get branch up to latest

---------

Below is a screenshot of what the updated folder structure would look like:

_folder_structure

### Next steps

- can add new submodules, I'm thinking to add a new one for a new server
- can start splitting existing code into submodules (https://github.com/triplea-game/triplea/issues/2848)

### CODE REVIEW

**Comment by RoiEXLab** [comment]
Hmm I'm not sure what exactly you have in mind when this project will be in its final form.
Which folder structure are you thinking of when separate modules are being added? 
Having a "root folder" for each individual "module" with an individual gradle script?
I almost believe it's going to be easier to make use of the module functionality of java 9 when we're heading towards an incompatible release anyways.
This would still place strict restrictions on what packages can be used by others, but also keep the project structure relatively simple at no additional cost since the long term plan is to change the package structure anyways.

**Comment by DanVanAtta** [comment]
@ssoloff if you get a chance, any help to find out why the custom checkstyle scripts are not working, checkstyle is returning 'zero' violations successfully, would be really appreciated!

**Comment by ssoloff** [comment]
@DanVanAtta Sure, I can take a look tomorrow.

**Comment by DanVanAtta** [comment]
@ssoloff thanks for checking. Very interesting : )
I was *really* tired when I wrapped this up and shared, in my exhaustion I either forget that checkstyle worked, or simply did not try it from the top level.

not_success

This then leaves to determine why integ tests are not passing : )

**Comment by DanVanAtta** [comment]
@ssoloff thanks for checking. Very interesting : )
I was *really* tired when I wrapped this up and shared, in my exhaustion I either forget that checkstyle worked, or simply did not try it from the top level.

Though, if the `checkstyleMain` is run from sub-project, then the zero result success behavior can be observed:
not_success

This then leaves to determine why integ tests are not passing : )

**Comment by ssoloff** [comment]
@DanVanAtta My guess is you had previously run `checkstyleMain` from the root and have made no source changes between then and running `checkstyleMain` from the subproject.  The Gradle Checkstyle tasks won't re-run if their inputs (typically _.class_ files) have not changed.  Try running `../gradlew clean checkstyleMain` from the subproject and see if you get non-zero Checkstyle violations.

**Comment by DanVanAtta** [comment]
@ssoloff you rock, ty -  that was it.

**Comment by DanVanAtta** [comment]
@RoiEXLab  Benefits from sub-project are not quite the same as Java 9 modules. A key difference is in dependencies, each sub-project has its own set. For example, we could have lobby ship with only its dependencies and no Substance JAR, and the game would not ship with postgres JAR. This PR is very early, it's just to set a first structure where we can create a next sub-project.  I was actually planning on creating a new lobby next with benefit that no dependencies had to be shared between new and old code.

As part of this being early, the gradle script that existed was largely moved to the child project. Sub-projects are organized as root folders with build scripts in them, common components can be factored out and generally, it should be pretty clean with dependency lists being the main difference between them.

For me to summarize, sub-projects provides benefits that you cannot get with package structure alone. JDK9 modules will complement sub-projects quite nicely actually, as with that we could lock down the API's of 'lib' sub-projects to just a few API's : )    I think this use-case is largely what they had in mind with JDK9 modules even, to make is so lib packages could expose just specific API's and not API's that happen to be used across package boundaries within a module.

Hopefully that makes it a bit more clear, if not I'll note my next immediate goal would be to create a new deployable and I would not want to introduce new lib dependencies to existing game engine or lobby. Since goal is new code that is isolated from existing, package restructure does not help that and is really just not related. So I'm trying to convey that JDK9 modules and sub-projects have different benefits, that modules should be used within the packages of a sub-project to harden internal modules. The split of build script and library dependencies is meanwhile very useful.

**Comment by DanVanAtta** [comment]
Got the last kinks worked out from the build, it passes now, no longer a preview, looking for a review now : )

**Comment by ssoloff** [comment]
@DanVanAtta Did you happen to set up a Travis release branch for this PR to verify the `deploy` phase of the _.travis.yml_ script is working as expected?  It would be nice to know that there won't be any surprises requiring fire fighting after merging to `master`. :smile:

**Comment by DanVanAtta** [comment]
@ssoloff biggest concern is about merge accuracy, doing that for a 3rd time soon is getting risky. Latest merge to master was not good. I'll merge and if the artifact is not built i'll patch or roll back. That is easier fire fighting compared to getting a merge wrong.

@ssoloff will #2977 be ready soon? That is the only PR that I think makes sense waiting for as rest are not ready or preview.

Assigning to myself as this needs merge resolution cleanup.

**Comment by DanVanAtta** [comment]
(short answer: no) @ssoloff , but i'm pretty sure updating paths will be sufficient. I ran the install4j to make sure that works. I don't expect any surprises there, and again, the fire fighting is not so bad and less cost than otherwise.

**Comment by ssoloff** [comment]
@DanVanAtta #2977 is now ready for review.

**Comment by DanVanAtta** [comment]
@ssoloff @RoiEXLab I don't think we can let this hang out until the PR queue is empty. Source file conflicts are easy to resolve for anyone on either side of this. I am concerned that it'll be hard to line up my having a chance to resolved conflicts and have a reviewer merge with the PR queue being empty.

I'll suggest we give it 24 hours more, I should have a chance tomorrow night to resolve conflicts and would propose to merge then.

**Comment by ssoloff** [comment]
@DanVanAtta No objections.  I was under the impression you still had at least one outstanding issue in the TODO list (re: Checkstyle) before this was ready to merge (conflicts notwithstanding).  I was withholding a final review until that time.

My suggestion would be that you self-merge this PR when you know you'll have a window in which you can address any critical follow-up issues that may arise on `master`.

**Comment by DanVanAtta** [comment]
@ssoloff  Got lost in branches a bit, checkstyle staying in same location is done: 34a99d8

File: game-core/build.gradle [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+    remoteLibsDir = file('.remote-libs')
+    rootFilesDir = file("$buildDir/rootFiles")
+    schemasDir = file('config/triplea/schemas')
+
+    gameEnginePropertiesFile = file('game_engine.properties')
+    gameEnginePropertiesArtifactFile = file("$rootFilesDir/${gameEnginePropertiesFile.name}")
+}
+
+checkstyle  {
+    configFile = file("$rootProject.projectDir/config/checkstyle/checkstyle.xml")
```
Comment (ssoloff) [inline game-core/build.gradle]: Can this just be added to the existing `checkstyle` task configuration on line 331?  (Note that it must be defined before the `configProperties` property, which makes use of the `configFile` property.)

Comment (DanVanAtta) [reply game-core/build.gradle]: Looks to not be a problem, good recommendation: 01062ba

**Comment by ssoloff** [comment]
Checkstyle threshold update script is failing on `master`:

```
$ ./.travis/update_checkstyle_thresholds
Checking if Checkstyle thresholds should be updated for this build...
fatal: Cannot open 'gradle.properties': No such file or directory
Updating Checkstyle thresholds...
grep: build/reports/checkstyle/main.xml: No such file or directory
sed: can't read gradle.properties: No such file or directory
grep: build/reports/checkstyle/test.xml: No such file or directory
sed: can't read gradle.properties: No such file or directory
grep: build/reports/checkstyle/integTest.xml: No such file or directory
sed: can't read gradle.properties: No such file or directory
Checking for changes to 'gradle.properties'...
fatal: Cannot open 'gradle.properties': No such file or directory
  ...no changes detected.
```

Also, tags don't appear to be working: https://github.com/triplea-game/triplea/releases/

I'll take a look at the Checkstyle script.  @DanVanAtta, could you please look at the tags?

**Comment by ssoloff** [comment]
#3019 and #3020

**Pull request merged**